modules/Bio/EnsEMBL/Compara/RunnableDB/PrepareMaster/RenameGenome.pm

```diff
@@ -0,0 +1,159 @@
+
+=head1 DESCRIPTION
+
+Renames genome in master and previous databases, as well as in mlss config file, species tree file and genome
+dumps. Files that will stop working due to the renaming and cannot be modified, i.e. exonerate and sketch
+files, are deleted so they are properly regenerated by their corresponding pipeline.
+
+=over
+
+=item old_name
+
+Mandatory. Old genome (species) name.
+
+=item new_name
+
+Mandatory. New genome (species) name.
+
+=item master_db
+
+Mandatory. Alias of the compara master database.
+
+=item prev_dbs
+
+Mandatory. Alias(es) or alias pattern(s) of previous databases to be renamed as well.
+
+=item xml_file
+
+Mandatory. Path to the MLSS configuration XML file.
+
+=item species_tree
+
+Mandatory. Path to the species tree file.
+
+=item genome_dumps_dir
+
+Mandatory. Path to the root folder where the genome dumps are stored.
+
+=item sketch_dir
+
+Mandatory. Path to the root folder where the sketch files are stored.
+
+=back
+
+=head1 EXAMPLES
+
+    standaloneJob.pl Bio::EnsEMBL::Compara::RunnableDB::PrepareMaster::RenameGenome \
+        -compara_db $(mysql-ens-compara-prod-10-ensadmin details url jalvarez_prep_citest_master_for_rel_100) \
+        -old_name canis_familiaris -new_name perricus_bonicus -master_db compara_master -prev_dbs '*_prev' \
+        -xml_file $ENSEMBL_CVS_ROOT_DIR/ensembl-compara/conf/citest/mlss_conf.xml \
+        -species_tree $ENSEMBL_CVS_ROOT_DIR/ensembl-compara/conf/citest/species_tree.branch_len.nw \
+        -genome_dumps_dir /hps/nobackup2/production/ensembl/jalvarez/genome_dumps/ \
+        -sketch_dir /hps/nobackup2/production/ensembl/jalvarez/species_tree/citest_sketches/
+
+=cut
+
+package Bio::EnsEMBL::Compara::RunnableDB::PrepareMaster::RenameGenome;
+
+use warnings;
+use strict;
+
+use Bio::EnsEMBL::Hive::Utils;
+use Bio::EnsEMBL::Registry;
+
+use base ('Bio::EnsEMBL::Compara::RunnableDB::BaseRunnable');
+
+
+sub fetch_input {
+    my $self = shift;
+
+    # Get all the required compara DBAdaptors
+    my $db_list = $self->param_required('prev_dbs');
+    $db_list = [$db_list] unless ref($db_list);
+    push @{$db_list}, $self->param_required('master_db');
+    my %dba_hash = $self->get_all_compara_dbas($db_list);
+    $self->param('dba_hash', \%dba_hash);
+}
+
+
+sub run {
+    my $self = shift;
+
+    my $old_name = $self->param_required('old_name');
+    my $new_name = $self->param_required('new_name');
+    my $xml_file = $self->param_required('xml_file');
+    my $species_tree = $self->param_required('species_tree');
+    my $genome_dumps_dir = $self->param_required('genome_dumps_dir');
+    my $sketch_dir = $self->param_required('sketch_dir');
+
+    my $master_db = $self->param('master_db');
+    my $dba_hash = $self->param('dba_hash');
+    my $genome_db = $dba_hash->{$master_db}->get_GenomeDBAdaptor()->fetch_by_registry_name($old_name);
+
+    # We really need a transaction to ensure we are not screwing the databases
+    $self->call_within_transaction(sub {
```

Comment:
This starts a transaction on `$self->compara_dba` but we need it one on each `$dba` instead.